### DISCUSSION TITLE
Mamba error installing Perl for MOOSE

### DISCUSSION THREAD

**Discussion by rulisp** [idaholab/moose#26460] [Q&A Getting Started]
Hello,

When attempting to follow the conda (mamba) installation instructions [here](https://mooseframework.inl.gov/getting_started/installation/conda.html) I receive the following error:

error    libmamba Error opening for reading "/cluster/VAST/rulisp-lab/cpg/mamba/pkgs/perl-5.32.1-4_hd590300_perl5/info/index.json": No such file or directory
perl-5.32.1-4_hd590300_perl5.conda extraction failed
Found incorrect download: perl. Aborting

I also retrieved the specific conda file and attempted a direct installation of Perl as:
(moose) [rulisp@hellbender:temp]$ mamba install linux-64_perl-5.32.1-4_hd590300_perl5.conda

Downloading and Extracting Packages

didn't find info-linux-64_perl-5.32.1-4_hd590300_perl5 component in /cluster/VAST/rulisp-lab/cpg/mamba/pkgs/linux-64_perl-5.32.1-4_hd590300_perl5.conda

LookupError("didn't find info-linux-64_perl-5.32.1-4_hd590300_perl5 component in /cluster/VAST/rulisp-lab/cpg/mamba/pkgs/linux-64_perl-5.32.1-4_hd590300_perl5.conda")

Any suggestion would be much appreciated.

Thank you,

**Comment by GiudGiud** [comment]
Hello

Looks like a bad package. If it s not recovering by itself, and removing the package is not enough, do you have the option to uninstall conda and try again from scratch?

Guillaume

**Comment by rulisp** [comment]
Hello,

Unfortunately, I don't have install (root) privileges for the system so I'm mostly stuck with the versions of conda and mamba that the administrators provide.

Is there a way to install a different version of Perl that will still be compatible with MOOSE, but which doesn't have the missing info?

Paul Rulis, PhD
Physics and Astronomy Graduate Advisor
Division of Energy, Matter, and Systems (EMS)
School of Science and Engineering (SSE)
University of Missouri - Kansas City
250D Flarsheim Hall, 5110 Rockhill Road
Kansas City, MO 64110 USA
+1-816-235-5223 (Voice)
+1-816-235-5159 (FAX)
Directory Webpage<https://sse.umkc.edu/profiles/rulis-paul.html> | Research Webpage<https://info.umkc.edu/cpg/>
***@***.***

**Reply by GiudGiud** [reply]
So you should be able to download a local mamba on which you do have privileges, instead of using a shared mamba on which you will have to deal with frozen package versions I m guessing

Did you try the curl download we suggest on the getting started / linux installation page?

**Reply by GiudGiud** [reply]
these https://mooseframework.inl.gov/getting_started/installation/conda.html
though if you are on HPC you want to disable mamba and use the manual install instead

**Comment by rulisp** [comment]
Yes, I'm working on an HPC system. I was initially hoping to do a quick conda install for testing, but since it is taking more time than I thought, I'll probably just punt the issue to the system administrators to see if they can do the proper HPC install for me.

Thank you very much for your assistance!

Paul Rulis, PhD
Physics and Astronomy Graduate Advisor
Division of Energy, Matter, and Systems (EMS)
School of Science and Engineering (SSE)
University of Missouri - Kansas City
250D Flarsheim Hall, 5110 Rockhill Road
Kansas City, MO 64110 USA
+1-816-235-5223 (Voice)
+1-816-235-5159 (FAX)
Directory Webpage<https://sse.umkc.edu/profiles/rulis-paul.html> | Research Webpage<https://info.umkc.edu/cpg/>
***@***.***

**Reply by GiudGiud** [reply]
> Yes, I'm working on an HPC system.

Yeah there are a few options then.
- follow the linux install, which you can do yourself with your own conda, not the system one. But the performance will be much lower notably because MPI is not compiled for your HPC system
- follow the HPC system install, which you should also be able to do yourself, but if there are missing libraries / modules you may want to ask your HPC team to install those at the cluster level
- get your HPC folks to install MOOSE for you and make it available as a module if that's what's used on that system

**Reply by GiudGiud** [reply]
If you follow the 3rd root, they probably would install the combined executable in moose/modules.
If you want to add new objects (coding new ones) you ll need to use dynamic loading to load them

**Comment by rulisp** [comment]
Thank you very much for the guidance. I'll probably see if the HPC administrators can do the install for me so I can load it as a module.

Paul Rulis, PhD
Physics and Astronomy Graduate Advisor
Division of Energy, Matter, and Systems (EMS)
School of Science and Engineering (SSE)
University of Missouri - Kansas City
250D Flarsheim Hall, 5110 Rockhill Road
Kansas City, MO 64110 USA
+1-816-235-5223 (Voice)
+1-816-235-5159 (FAX)
Directory Webpage<https://sse.umkc.edu/profiles/rulis-paul.html> | Research Webpage<https://info.umkc.edu/cpg/>
***@***.***

**Comment by rulisp** [comment]
That would be using https://mooseframework.inl.gov/source/actions/DynamicObjectRegistrationAction.html ?

Paul Rulis, PhD
Physics and Astronomy Graduate Advisor
Division of Energy, Matter, and Systems (EMS)
School of Science and Engineering (SSE)
University of Missouri - Kansas City
250D Flarsheim Hall, 5110 Rockhill Road
Kansas City, MO 64110 USA
+1-816-235-5223 (Voice)
+1-816-235-5159 (FAX)
Directory Webpage<https://sse.umkc.edu/profiles/rulis-paul.html> | Research Webpage<https://info.umkc.edu/cpg/>
***@***.***

**Reply by GiudGiud** [reply]
exactly

**Comment by rulisp** [comment]
Great, thank you!

Paul Rulis, PhD
Physics and Astronomy Graduate Advisor
Division of Energy, Matter, and Systems (EMS)
School of Science and Engineering (SSE)
University of Missouri - Kansas City
250D Flarsheim Hall, 5110 Rockhill Road
Kansas City, MO 64110 USA
+1-816-235-5223 (Voice)
+1-816-235-5159 (FAX)
Directory Webpage<https://sse.umkc.edu/profiles/rulis-paul.html> | Research Webpage<https://info.umkc.edu/cpg/>
***@***.***
